### PR TITLE
[rtabmap] Fix platform and linkage support

### PR DESCRIPTION
**Pull request by dg0yt**
`rtabmap-res_tool` is moved to a separate port because it is a host dependency, and actual dependencies of `rtabmap` are heavy and non opt-out. Only minimal patching needed.

CC @matlabbe.

### CODE REVIEW

**Comment by dg0yt** [comment]
~~~
/mnt/vcpkg-ci/installed/x64-linux/share/vtk/vtk-prefix.cmake(1):  set(_vtk_module_import_prefix ${CMAKE_CURRENT_LIST_DIR} )
/mnt/vcpkg-ci/installed/x64-linux/share/vtk/vtk-prefix.cmake(2):  get_filename_component(_vtk_module_import_prefix ${_vtk_module_import_prefix} DIRECTORY )
/mnt/vcpkg-ci/installed/x64-linux/share/vtk/vtk-prefix.cmake(3):  get_filename_component(_vtk_module_import_prefix ${_vtk_module_import_prefix} DIRECTORY )
/mnt/vcpkg-ci/installed/x64-linux/share/vtk/vtk-config.cmake(110):  set(${CMAKE_FIND_PACKAGE_NAME}_PREFIX_PATH ${_vtk_module_import_prefix} )
/mnt/vcpkg-ci/installed/x64-linux/share/vtk/vtk-config.cmake(112):  unset(_vtk_module_import_prefix )
/mnt/vcpkg-ci/installed/x64-linux/share/vtk/vtk-config.cmake(113):  list(INSERT CMAKE_PREFIX_PATH 0 ${${CMAKE_FIND_PACKAGE_NAME}_PREFIX_PATH} )
~~~
Ouch. vtk breaking vcpkg CMAKE_PREFIX_PATH for debug configurations. (Because of vcpkg moving cmake config.)
I think we can just remove this line. CC @Neumann-A

**Comment by Neumann-A** [comment]
That seems to be https://github.com/microsoft/vcpkg/pull/37119/commits/95052b826401ecbcde023f29b8fab7c29f1358a5 what I fixed in the 5.12 PR

**Comment by dg0yt** [comment]
The vtk patch is part of  #37119.

File: ports/rtabmap/link-keywords.patch [hunk 1]
```diff
@@ -0,0 +1,12 @@
+diff --git a/corelib/src/CMakeLists.txt b/corelib/src/CMakeLists.txt
+index 6a4a9ab..85ede4e 100644
+--- a/corelib/src/CMakeLists.txt
++++ b/corelib/src/CMakeLists.txt
+@@ -167,7 +167,6 @@ SET(LIBRARIES
+ # Issue that qhull dependency uses optimized and debug keywords,
+ # which are converted to \$<\$<NOT:\$<CONFIG:DEBUG>> and \$<\$<CONFIG:DEBUG>
+ # in RTABMap_coreTargets.cmake (not sure why?!).
+-list(REMOVE_ITEM PCL_LIBRARIES "debug" "optimized")
```
Comment (dg0yt) [inline ports/rtabmap/link-keywords.patch:9]: This caused harmful mixing of release libs and debug libs.

**Comment by Cheney-W** [comment]
In local testing of the `rtabmap` feature, the failure occurred due to the inability to install `liblzma`, which is a known issue.

File: ports/rtabmap/portfile.cmake [hunk 2]
```diff
@@ -33,59 +29,62 @@ vcpkg_cmake_configure(
...
         ${FEATURE_OPTIONS}
         -DBUILD_AS_BUNDLE=OFF
         -DBUILD_EXAMPLES=OFF
-        -DWITH_ORB_OCTREE=ON
-        -DWITH_TORCH=OFF
-        -DWITH_PYTHON=OFF
-        -DWITH_PYTHON_THREADING=OFF
-        -DWITH_PDAL=OFF
+        -DRTABMAP_QT_VERSION=6
+        "-DRTABMAP_RES_TOOL=${CURRENT_HOST_INSTALLED_DIR}/tools/rtabmap-res-tool/rtabmap-res_tool${VCPKG_TARGET_EXECUTABLE_SUFFIX}"
```
Comment (BillyONeal) [inline ports/rtabmap/portfile.cmake:37]: Review, no change requested. This is a sort of the options list. These are the actual changes.

* `-DRTABMAP_QT_VERSION=6` added
* `"-DRTABMAP_RES_TOOL=...` added
* `-DWITH_ORB_OCTREE=ON` gained a comment, `   # GPLv3`
* `-DWITH_ORB_SLAM=OFF` added

**Review by BillyONeal — APPROVED** [review]
Thanks for the fixes!

**Pull request merged**